Add Getter's Keys operation. (#825)
open-telemetry/opentelemetry-specification: specification/context/api-propagators.md

Returns list of fields that will be used by the `HttpTextPropagator`.

Observe that some `Propagator`s may define additional fields with variable names, such

> as fields with different names simply sharing the `jaeger-` prefix.

Reviewer: please add a reference on some documentation about `jaeger-` keys.

Reviewer 2: It's not actually `jaeger-`, but `uberctx-{user-defined-key}`. In other words, the tracer does not know ahead of time the suffixes of the relevant headers, only that they all start with `uberctx-`, thus it must use an iterator to discover them.

https://www.jaegertracing.io/docs/1.18/client-libraries/#baggage

Reviewer: @carlosalberto can you add this link to the spec for a quick reference on why it's needed

> as fields with different names simply sharing the `jaeger-` prefix.

Reviewer: What should propagator return in this case? Can you please finish this sentence with the actionable suggestion.

`Getter` allows a `HttpTextPropagator` to read propagated fields from a carrier.

> One of the ways to implement it is `Getter` class with `Get` and `Keys` methods

Reviewer: what is another way to implement it? Either this PR is incomplete or it shouldn't say "one of the ways"

Author: As this existed already, I suggest we do a follow up for this part (most likely we won't use the `One of the ways to implement it` part ;) )

as described below.

##### Keys

> The Keys function MUST return the list of all the keys in the carrier.

Reviewer: ```suggestion
The `Keys` function MUST return the list of all the keys in the carrier.
```

Required arguments:

> - The carrier of the propagation  fields, such as an HTTP request.

Reviewer: nit:

```suggestion
- The carrier of the propagation fields, such as an HTTP request.
```

> The Keys function can be called by `Propagator`s using variable key names in order to
> iterate over the entire keys in the specified carrier. For example, it can be used to
> detect all keys starting with the prefix `jaeger-`.

Reviewer: Small rephrasing suggestion:

```suggestion
The `Keys` function can be called by `Propagator`s which are using variable key names in order to
iterate over all keys in the specified carrier. For example, it can be used to
detect all keys starting with the prefix `jaeger-`.
```

`Getter` allows a `HttpTextPropagator` to read propagated fields from a carrier.

> One of the ways to implement it is `Getter` class with `Get` and `Keys` methods
> as described below.

Reviewer: this is how "one of the way" sentence may be improved: 

```suggestion
One of the ways to implement it is `Getter` class with `Get` and `Keys` methods
as described below. Languages may decide on alternative implementations and 
expose corresponding methods as delegates or other ways.
```
